src/gather/api.py

```diff
@@ -53,48 +97,19 @@ class Collector(object):
 
     depth = attr.ib(default=1)
 
-    @staticmethod
-    def one_of(_registry, _effective_name, objct):
-        """Assign one of the possible options.
+    one_of = staticmethod(_one_of)
 
-        When given as a collection strategy to :code:`collect`,
-        will assign one of the options to a name in case more
-        than one item is registered to the same name.
+    all = staticmethod(_all)
 
-        This is the default.
-        """
-        return objct
-
-    @staticmethod
-    def all(registry, effective_name, objct):
-        """Assign all of the possible options.
-
-        Collect all registered items into a set,
-        and assign that set to a name. Note that
-        even if only one item is assigned to a name,
-        that name will be assigned to a set of length 1.
-        """
-        myset = registry.get(effective_name, set())
-        myset.add(objct)
-        return myset
-
-    @staticmethod
-    def exactly_one(registry, effective_name, objct):
-        """Raise an error on colliding registration.
-
-        If more than one item is registered to the
-        same name, raise a :code:`GatherCollisionError`.
-        """
-        if effective_name in registry:
-            raise GatherCollisionError("Attempt to double register",
-                                       registry, effective_name, objct)
-        return objct
+    exactly_one = staticmethod(_exactly_one)
 
     def register(self, name=None, transform=lambda x: x):
-        """Register
+        """
+        Register
 
-        :param name: optional. Name to register as (default is name of object)
-        :param transform: optional. A one-argument function. Will be called,
+        Args:
+            name (str): optional. Name to register as (default is name of object)
```

Comment:
Name to register the _callable_ as.

-Or-

Name under which to register the callabe.